frontend/src/metabase/auth/actions.ts

```diff
@@ -66,27 +66,22 @@ export const loginGoogle = createThunkAction(
 );
 
 export const LOGOUT = "metabase/auth/LOGOUT";
-export const logout = createThunkAction(
-  LOGOUT,
-  (redirectUrl: string, isSessionAlreadyExpired: boolean) => {
-    return async (dispatch: any) => {
-      if (!isSessionAlreadyExpired) {
-        await deleteSession();
-      }
-      await dispatch(clearCurrentUser());
-      await dispatch(refreshLocale());
-      trackLogout();
+export const logout = createThunkAction(LOGOUT, (redirectUrl: string) => {
+  return async (dispatch: any) => {
+    await deleteSession();
+    await dispatch(clearCurrentUser());
+    await dispatch(refreshLocale());
+    trackLogout();
 
-      let loginUrl = "/auth/login";
-      if (redirectUrl) {
-        loginUrl += `?redirect=${encodeURIComponent(redirectUrl)}`;
-      }
+    let loginUrl = "/auth/login";
+    if (redirectUrl) {
+      loginUrl += `?redirect=${encodeURIComponent(redirectUrl)}`;
+    }
 
-      dispatch(push(loginUrl));
-      window.location.reload(); // clears redux state and browser caches
-    };
-  },
-);
+    dispatch(push(loginUrl));
+    window.location.reload(); // clears redux state and browser caches
+  };
+});
```

Comment:
@alxnddr This is the only change on the FE. I made sure the session is deleted every time the session middleware calls `logout`. Before, we were assuming the `metabase.SESSION` cookie had expired whenever the `metabase.TIMEOUT` cookie had expired. This is no longer the case. Now, if the `metabase.TIMEOUT` cookie expires, we need to clear the `metabase.SESSION` cookie.